pythonzm_Ops_task_views_41.py

```
@@ -0,0 +1,412 @@
+def role_edit(request):
+    if request.method == 'POST':
+        content = request.POST.get('content')
+        relative_path = request.POST.get('relative_path')
+        p_name = request.POST.get('p_name')
+        name = request.POST.get('name')
+
+        try:
+            if relative_path:
+                with open(os.path.join(settings.ANSIBLE_ROLE_PATH, relative_path), 'w') as f:
+                    f.write(content)
+                return JsonResponse({'code': 200, 'msg': '修改完成！'})
```

Ruff (0.13.1)

String contains ambiguous `！` (FULLWIDTH EXCLAMATION MARK). Did you mean `!` (EXCLAMATION MARK)?

(RUF001)